File: median-of-two-sorted-arrays.cpp
//SOLUTION FOR LEETCODE problem of finding median fo two sorted arrays 
class Solution {
public:
    double findMedianSortedArrays(vector<int>& nums1, vector<int>& nums2)
    {
        int m = nums1.size();
        int n = nums2.size();
        int N = m + n;
        vector<int> merged(N, 0);

        int i, j, k;
        i = j = k = 0;
        while(i<m && j<n) {
            if(nums1[i] < nums2[j])
                merged[k++] = nums1[i++];
            else
                merged[k++] = nums2[j++];
        }
        while(i<m)
            merged[k++] = nums1[i++];
        while(j<n) 
            merged[k++] = nums2[j++];

        if(N&1) 
            return merged[N/2];
        else
        return (merged[(N/2)-1] + merged[N/2]) / 2.0;
    }
};

File: median-of-two-sorted-arrays.py
import math
class Solution(object):
    def findMedianSortedArrays(self, nums1, nums2):
        """
        :type nums1: List[int]
        :type nums2: List[int]
        :rtype: float
        """
        nums3=nums1+nums2
        nums3.sort()
        if len(nums3)%2==0:
            
            return float((nums3[len(nums3)/2]+nums3[len(nums3)/2-1]))/2
        else:
            return nums3[int(math.ceil(len(nums3)/2))]
